File: rate-limiter/typescript/src/Rule.ts
import { LimiterRuleInvalidError } from './errors.js';

export class Rule {
  readonly maxRequests: number;
  readonly windowMs: number;

  constructor(maxRequests: number, windowMs: number) {
    if (maxRequests <= 0) {
      throw new LimiterRuleInvalidError('Max requests must be positive');
    }
    if (windowMs <= 0) {
      throw new LimiterRuleInvalidError('Window must be positive');
    }
    this.maxRequests = maxRequests;
    this.windowMs = windowMs;
  }
}

File: rate-limiter/csharp/src/RateLimiter/Rule.cs
namespace RateLimiter;

public readonly record struct Rule
{
    public int MaxRequests { get; }
    public TimeSpan Window { get; }

    public Rule(int maxRequests, TimeSpan window)
    {
        if (maxRequests <= 0)
            throw new LimiterRuleInvalidException("Max requests must be positive");
        if (window <= TimeSpan.Zero)
            throw new LimiterRuleInvalidException("Window must be positive");

        MaxRequests = maxRequests;
        Window = window;
    }
}
